src/eigen.jl

```diff
@@ -5,17 +5,23 @@ import Base: +
 """
     eigen(K::AbstractKroneckerProduct)
 
-Wrapper around `eigen` from the `LinearAlgebra` package. Performs Eigenvalue decompositon
-on the matrices of a `AbstractKroneckerProduct` instances and returns an
-`Eigen` type. The functions `size`, `\\`, `inv`, `det`,
-and `logdet` are overloaded to efficiently work with this type.
+Wrapper around `eigen` from the `LinearAlgebra` package.
+If the matrices of an `AbstractKroneckerProduct` instance are square,
+performs Eigenvalue decompositon on them and returns an `Eigen` type.
+Otherwise, it collects the instance and runs eigen on the full matrix.
```

Comment:
But would it then not be possible to use SVD on both matrices and combine the eigenvalues?

Reply:
You are right, computing the SVD will be much more efficient than collecting the matrix and running `eigen` in this case. However, that would give us the [SVD of the Kronecker product](https://www.cs.cornell.edu/cv/ResearchPDF/KSVD.pdf), not the eigen-decomposition. So I would recommend leaving `eigen` as is, and extend `svd` instead. I will look into that!

If we wanted to make `eigen` efficient in this case, I believe adapting the QR algorithm for Kronecker matrices has a chance of working out well. But I haven't looked at it closer.

Reply:
Let me know what you think, and I'll make the other change later!

Reply:
And I don't know what you mean with bumping the version number. Do you mean somewhere in the code or a release on GitHub?